vm/http_test.go

```diff
@@ -91,17 +91,17 @@ func TestHTTPObjectFail(t *testing.T) {
 		require "net/http"
 
 		Net::HTTP.get("http://127.0.0.1:3000/error")
-		`, "InternalError: 404 Not Found", 4},
+		`, "HTTPError: Non-200 response, 404 Not Found (404)", 4},
 		{`
 		require "net/http"
 
 		Net::HTTP.post("http://127.0.0.1:3000/error", "text/plain", "Let me down")
-		`, "InternalError: 404 Not Found", 4},
+		`, "HTTPError: Non-200 response, 404 Not Found (404)", 4},
```

Comment:
@ear7h I think we shouldn't raise error just because the response code is not 200. It's very normal to get response code other then 2xx. So it might be better to let user know the situation and handle that themselves.

Reply:
This class method is more more of a 'quick and dirty' approach which only returns the response body directly as a string. This behavior is consistent with ruby's `Net::HTTP.get` method (ruby does not implement a quick post method, only post form). 
When using the client with `Net::HTTP.start` errors are not thrown on non-200 responses.

Reply:
Ok, but you need to clearly document this for others to know the result and reason.